### ISSUE TITLE
Custom "From" address does not work for drafts

### ISSUE THREAD

**Issue by krza** [speedball2001/identitychooser-mx#64]
Scenario:

1. I create a new email through answering a received one.
2.  I am asked for an identity which I choose.
3. I change the "From" address to a custom address (which is not defined in the thunderbird identities)
4. I edit the email and save it to the drafts.
5. I see this email in the drafts folder with the correct custim "From" address.
6. I open the email from the drafts in order to continue editing.
7. I am asked again to choose an identity – silly because it was selected already when starting the email but processwise it´s clear that the request comes again).
8. I try to change the "From" address again but now this item is grayed out, so I cannot change it again.

When I try it with another draft which has no custom address set, all is the same except from the last bullet point: Then I can change the "From" address. But if I do, save it again and open it again, it´s the same issue.

Only possible workarund at the moment: To reply to the original email again (if it still exists) and copy/paste the edited content from the drafts over to it, then remove the draft.

**Comment by bvandergiessen** [comment]
I would like to file point 7 as a usability issue on its own. As it is related, I'm replying to this one (let me know if I should enter it separately).

**Steps to reproduce:**

1. Create a message with one of your accounts as the sender, save it as a draft, close it
2. Select the draft and click <kbd>Edit</kdb>

**Expected behavior:**
The draft's sender remains the same it already was

**Actual behavior:**
The extension asks you which account should be the sender.

In other words, the account-selection popup should not appear when opening an existing draft for editing.

**Comment by speedball2001** [comment]
I made it configurable if Identity Chooser should ask for the identity when opening a draft.

Please test the attache version 4.2beta1 and let me know if it works for you.

[identitychooser.zip](https://github.com/speedball2001/identitychooser-mx/files/13821428/identitychooser.zip)

**Comment by bvandergiessen** [comment]
Works here!

**Comment by krza** [comment]
Stupid question ... how do I try it with the zip file? I thought that I would find the extension files somewhere on the disk but there´s just the xpi file. I don´t even find where its settings are stored. My approach was to backup the current files, replace them by the beta files and try it out. Afterwards I would have restored the previous content in case my settings had been overwritten by the beta files.

**Comment by speedball2001** [comment]
> Stupid question ... how do I try it with the zip file? I thought that I would find the extension files somewhere on the disk but there´s just the xpi file. I don´t even find where its settings are stored. My approach was to backup the current files, replace them by the beta files and try it out. Afterwards I would have restored the previous content in case my settings had been overwritten by the beta files.

You go to Addons & Themes, click the cog wheel to open a menu and then choose "Install from file"

**Comment by krza** [comment]
Okay, it worked and it had preseved the settings (thus selection and order of addresses). Thanks.

Test result wrt. the actual issue: 

Yes, it (usually) works now. At least partly, even if this a rather formal aspect. What I found now is the following:

1. When opening a draft, you can change the From address now.
2. If you close the dialog and open it again to change the address again, it´s grayed out again though.
3. This does not hurt too much because closing the draft and opening it again helps because the very first attempt seems to be always successful.
4. I found at least one old draft which does not work though. You cannot change the address at all. It is the only draft from one specific account.
5. I found some drafts from another account which work but still provoke the identity selection dialog. Afterwards it´s the same as described in #1 and #2 above.
6. Other old ones from other accounts worked normally (as in #1 and #2) on the other hand.

Conclusion:

It´s no blocking issue anymore because the workaround #3 should work for almost all drafts. The exceptions from #4 should be seldom.

Only in case you would like to dive deeper you might provide a version which writes debug data to somewhere from where I could copy it from. The main question is why there are 3 different behaviors in the same client and session. Not sure if the root cause could be seen on the add-on level (or if it´s hidden deeper in TB itself). On the other hand it´s actually not too important anymore because of the workaround and because it´s a rather special use case anyway.

**Comment by speedball2001** [comment]
In my (albeit limited) testing, Thunderbird allows to set a custom from address only once.

If I disable Identity Chooser (in fact, I tested it with an installation that doesn't even have Identity Chooser installed), I create an email, set the from address to a custom address, save the email as a draft and close the window. If I now edit the draft, the menu item to set the from address to a custom address is greyed out. 

For me this means that the greyed out custom from address is a Thunderbird issue not an Identity Chooser issue.

**Comment by krza** [comment]
Interesting. Thanks for the details. However, I still have the feeling that the behavior has changed somehow ... hold on ... you are right ... saving the draft is the point, _after_ changing the address. If you just change the address but nothing else and close the draft, TB does not even ask to save this change because it´s ignored (and in fact does not happen).

Okay, I cannot recall why exactly I initially came to the conclusion that the identity chooser was the changing factor. Probably by cance because the trigger condition is kind of tricky. However, even if that was not solved, we now know the critical conditions and can try to prevent them. And ... there is an additional setting now to omit the chooser dialog for drafts ;)
